docs/overview.md

```diff
@@ -1,4 +1,31 @@
 # The Component Model
 
-asdf
+The Component Model for bots is our framework for sharing and reusing bot functionality - through new bots with templates, using portions of bots in the form of packages, or as complete bots with skills.
 
+To work with components from Composer while this feature is in preview, you'll need to enable this feature in your Composer settings menu - Settings > Preview Features > New Creation Experience.
+
+## Table of Contents
+
+1. [Overview](~/docs/overview.md)
+2. [Extending your bot using packages](~/docs/extending-with-packages.md)
+3. [Extending your bot with code](~/docs/extending-with-code.md)
+4. [Creating your own packages](~/docs/creating-packages.md)
+5. [Creating your own templates](~/docs/creating-templates.md)
+
+## Adaptive Runtime
+
+At the core of the component model is the adaptive runtime - an extensible, configurable runtime that is treated as a black box to bot developers and taken as a dependency. The runtime provides extensibility points to add additional functionality by importing packages, connection to skills, or adding your own coded extensions.
+
+## Packages
+
+Packages are bits of a bot you want to share/import like declarative dialog assets, coded dialogs, custom adapters, middleware or custom actions. They are just regular NuGet or npm packages (depending on the code language for your bot). You'll use the Bot Framework CLI tool to merge the package's declarative contents with your bot (package management in Composer will handle this for you, or you could do it yourself using the Bot Framework CLI tool). They can be made up of any combination of declarative assets (.dialog, .lu, .lg, .qna files), coded extensions (custom actions, middleware, adapters), or just plain old package libraries.
```

Comment:
I would remove middleware for now and stick to custom actions and adapters. We still don't have a solidified story for middleware.